src/sentry/tagstore/snuba/backend.py

```diff
@@ -16,69 +17,363 @@
+            issue: ObjectWrapper({
+                'times_seen': count,
+                'key': key,
+                'value': value,
+                'group_id': issue,
+            }) for issue, count in six.iteritems(result)}
 
     def get_group_tag_value_count(self, project_id, group_id, environment_id, key):
-        pass
+        start, end = self.get_time_range(7)
+        tag = 'tags[{}]'.format(key)
+        filters = {
+            'project_id': [project_id],
+            'environment': [environment_id],
+            'issue': [group_id],
+        }
+        conditions = [[tag, '!=', '']]
+        aggregations = [['count', '', 'count']]
 
-    def get_group_tag_values_for_users(self, event_users, limit=100):
-        pass
+        return snuba.query(start, end, [], conditions, filters, aggregations)
 
     def get_top_group_tag_values(self, project_id, group_id, environment_id, key, limit=3):
-        pass
+        start, end = self.get_time_range(7)
+        tag = 'tags[{}]'.format(key)
+        filters = {
+            'project_id': [project_id],
+            'environment': [environment_id],
+            'issue': [group_id],
+        }
+        conditions = [[tag, '!=', '']]
+        aggregations = [
+            ['count', '', 'count'],
+            ['min', SEEN_COLUMN, 'first_seen'],
+            ['max', SEEN_COLUMN, 'last_seen'],
+        ]
+
+        result = snuba.query(start, end, [tag], conditions, filters,
+                             aggregations, limit=limit, orderby='-count')
+
+        return [ObjectWrapper({
+            'times_seen': val['count'],
+            'first_seen': val['first_seen'],
+            'last_seen': val['last_seen'],
+            'key': key,
+            'value': name,
+            'group_id': group_id,
+        }) for name, val in six.iteritems(result)]
+
+    def get_group_tag_keys_and_top_values(self, project_id, group_id, environment_id, user=None):
+        from sentry import tagstore
+        start, end = self.get_time_range()
+        filters = {
+            'project_id': [project_id],
+            'environment': [environment_id],
+            'issue': [group_id],
+        }
+        aggregations = [
+            ['topK(10)', 'tags.value', 'top'],
+            ['count', '', 'count'],
+            ['uniq', 'tags.key', 'uniq'],
+        ]
+        results = snuba.query(start, end, ['tags.key'], None, filters,
+                              aggregations, arrayjoin='tags')
+
+        return [{
+            'id': key,
+            'name': tagstore.get_tag_key_label(key),
+            'key': tagstore.get_standardized_key(key),
+            'uniqueValues': res['uniq'],
+            'totalValues': res['count'],
+            'topValues': [{
+                'id': val,
+                'name': tagstore.get_tag_value_label(key, val),
+                'key': tagstore.get_standardized_key(key),
+                'value': val,
+                # TODO we don't know any of these without more queries
```

Comment:
Hm, I wonder if they're there just to be serialized "just cause" and are never used?

Reply:
Will ignore them for now and if they are needed I'll find out.